app/models/manageiq/providers/foreman/provider.rb

```diff
@@ -25,6 +25,56 @@ class ManageIQ::Providers::Foreman::Provider < ::Provider
   validates :name, :presence => true, :uniqueness => true
   validates :url,  :presence => true
 
+  def self.params_for_create
+    @params_for_create ||= {
+      :title  => "Configure AWS",
+      :fields => [
+        {
+          :component  => "text-field",
+          :name       => "endpoints.default.base_url",
+          :label      => "URL",
+          :isRequired => true,
+          :validate   => [{:type => "required-validator"}]
+        },
+        {
+          :component  => "text-field",
+          :name       => "endpoints.default.username",
+          :label      => "User",
+          :isRequired => true,
+          :validate   => [{:type => "required-validator"}]
+        },
+        {
+          :component  => "text-field",
+          :name       => "endpoints.default.password",
+          :label      => "Password",
+          :type       => "password",
+          :isRequired => true,
+          :validate   => [{:type => "required-validator"}]
+        },
+        {
+          :component => "checkbox",
+          :name      => "endpoints.default.verify_ssl",
+          :label     => "Verify SSL"
+        }
+      ]
+    }.freeze
+  end
+
+  # Verify Credentials
+  # args: {
+  #  "endpoints" => {
+  #    "default" => {
+  #      "base_url" => nil,
+  #      "username" => nil,
+  #      "password" => nil,
+  #      "verify_ssl" => nil
+  #    }
+  #  }
+  # }
+  def self.verify_credentials(args)
+    raw_connect(*args.dig("endpoints", "default")&.values_at("base_url", "username", "password", "verify_ssl"))
```

Comment:
Needs to return boolean.

Reply:
Also is verify_ssl supposed to be boolean for one of OpenSSL::VERIFY_*?

Reply:
Also, also...needs to try_decrypt the password.  I've noticed in other providers that's done in the raw_connect method itself, which seems like a nicer place since it allows decrypting from other entrypoints as well.

Reply:
Yup, was in WIP until I got it to verify the connection not just return it even if everything was wrong.

Addressed all of those PTAL